### ISSUE TITLE
Add support for instrumentation library scope

### ISSUE THREAD

**Issue by jack-berg** [open-telemetry/opentelemetry-java#4190]
To reflect spec changes from [#2276](https://github.com/open-telemetry/opentelemetry-specification/pull/2276)

- [ ] Add `InstrumentationScopeInfo`
- [ ] Adjust `TracerProvider` docs to reflect that `name` is instrumentation scope name
- [ ] Adjust `MeterProvider` docs to reflect that `name` is instrumentation scope name
- [ ] Adjust `LogEmitterProvider` to reflect that `name` is instrumentation scope name
- [ ] Adjust `ReadableSpan`, `SpanData` to include accessor to `InstrumentationScopeInfo`, deprecate `#getInstrumentationLibraryInfo()`
- [ ] Adjust `MetricData` to include accessor to `InstrumentationScopeInfo`, deprecate `#getInstrumentationLibraryInfo()`
- [ ] Adjust `LogData` to include accessor to `InstrumentationScopeInfo`, deprecate `#getInstrumentationLibraryInfo()`
- [ ] Adjust zipkin exporter to export `otel.scope.name`, `otel.scope.version`
- [ ] Adjust jaeger exporter to export `otel.scope.name`, `otel.scope.version`

**Comment by jkwatson** [comment]
for zipkin/jaeger, do we still export the old field name, too? Will that possibly break people?

**Comment by jack-berg** [comment]
> for zipkin/jaeger, do we still export the old field name, too? Will that possibly break people?

Yes the spec says to export the old field name and the new name for compatibility
